File: alpha/js/__tests__/testAssignment.js
var Assignment = require('../es6/assignment');
var UniformChoice = require('../es6/ops/random').UniformChoice;

var tester_unit = '4';
var tester_salt = 'test_salt';


describe('Test the assignment module', function() {
	it('Should set constants correctly', function() {
		var a = new Assignment(tester_salt);
		a.set('foo', 12);
		expect(a.get('foo')).toBe(12);
	});
	it('Should work with uniform choice', function() {
		var a = new Assignment(tester_salt);
		var choices = ['a', 'b'];
		a.set('foo', new UniformChoice({'choices': choices, 'unit': tester_unit}));
		a.set('bar', new UniformChoice({'choices': choices, 'unit': tester_unit}));
		a.set('baz', new UniformChoice({'choices': choices, 'unit': tester_unit}));

		expect(a.get('foo')).toEqual('b');
		expect(a.get('bar')).toEqual('a');
		expect(a.get('baz')).toEqual('a');
	});

	it('Should work with overrides', function() {
    var a = new Assignment(tester_salt);
    a.set_overrides({'x': 42, 'y': 43})
    a.set('x', 5);
    a.set('y', 6);
    expect(a.get('x')).toEqual(42);
    expect(a.get('y')).toEqual(43);
	});
});

File: python/planout/test/test_assignment.py
# Copyright (c) 2014, Facebook, Inc.
# All rights reserved.
#
# This source code is licensed under the BSD-style license found in the
# LICENSE file in the root directory of this source tree. An additional grant
# of patent rights can be found in the PATENTS file in the same directory.

import unittest

from planout.assignment import Assignment
from planout.ops.random import UniformChoice


class AssignmentTest(unittest.TestCase):
    tester_unit = 4
    tester_salt = 'test_salt'

    def test_set_get_constant(self):
        a = Assignment(self.tester_salt)
        a.foo = 12
        self.assertEqual(a.foo, 12)

    def test_set_get_uniform(self):
        a = Assignment(self.tester_salt)
        a.foo = UniformChoice(choices=['a', 'b'], unit=self.tester_unit)
        a.bar = UniformChoice(choices=['a', 'b'], unit=self.tester_unit)
        a.baz = UniformChoice(choices=['a', 'b'], unit=self.tester_unit)
        self.assertEqual(a.foo, 'b')
        self.assertEqual(a.bar, 'a')
        self.assertEqual(a.baz, 'a')

    def test_overrides(self):
        a = Assignment(self.tester_salt)
        a.set_overrides({'x': 42, 'y': 43})
        a.x = 5
        a.y = 6
        self.assertEqual(a.x, 42)
        self.assertEqual(a.y, 43)

    def test_custom_salt(self):
        a = Assignment(self.tester_salt)
        custom_salt = lambda x,y: '%s-%s' % (x,y)
        a.foo = UniformChoice(choices=list(range(8)), unit=self.tester_unit)
        self.assertEqual(a.foo, 7)

if __name__ == '__main__':
    unittest.main()
